custom-vu/src/main/kotlin/jces1209/vu/action/WorkOnSprint.kt

```diff
@@ -0,0 +1,87 @@
+import jces1209.vu.page.boards.view.ScrumBoardPage
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+
+class WorkOnSprint(
+    private val meter: ActionMeter,
+    private val jiraTips: JiraTips,
+    private val scrumBoardsMemory: SeededMemory<BoardPage>
+) : Action {
+    private val logger: Logger = LogManager.getLogger(this::class.java)
+
+    override fun run() {
+        val scrumBoard = scrumBoardsMemory.recall()
+        if (scrumBoard == null) {
+            logger.debug("I cannot recall scrum board, skipping...")
+            return
+        }
+
+        if (scrumBoard is ScrumBoardPage) {
+            val sprintBoardComponent = scrumBoard.goToBacklog()
+
+            meter.measure(MeasureType.SPRINT_CREATE) {
+                sprintBoardComponent.createSprint()
+            }
+
+            jiraTips.closeTips()
+
+            if (sprintBoardComponent.backlogIssuesNumber() > 0) {
+                meter.measure(MeasureType.SPRINT_MOVE_ISSUE) {
+                    sprintBoardComponent.moveIssueToSprint()
+                }
+
+                // Move additional times for reordering
+                repeat(10) {
+                    sprintBoardComponent.moveIssueToSprint()
+                }
+
+                if (sprintBoardComponent.isStartSprintEnabled()) {
+                    meter.measure(MeasureType.SPRINT_START_SPRINT) {
+                        meter.measure(MeasureType.SPRINT_START_SPRINT_EDITOR) {
+                            sprintBoardComponent.openStartSprintPopUp()
+                        }
+                        meter.measure(MeasureType.SPRINT_START_SPRINT_SUBMIT) {
+                            sprintBoardComponent.submitStartSprint()
+                        }
+                    }
+                } else {
+                    logger.debug("Can't start sprint - the option is disabled")
+                }
+            } else {
+                logger.debug("Scrum backlog doesn't contain issues in backlog")
+            }
+
+            val sprintPage = scrumBoard.goToActiveSprint()
+
+            if (sprintPage.maxColumnIssuesNumber() > 1) {
+                meter.measure(MeasureType.SPRINT_REORDER_ISSUE) {
+                    sprintPage.reorderIssue()
+                }
+            } else {
+                logger.debug("Scrum backlog doesn't contain enough issues to make reorder")
+            }
+
+            if (sprintPage.isCompleteButtonPresent()) {
+                meter.measure(MeasureType.SPRINT_COMPLETE) {
+                    try {
+                        sprintPage.completeSprint()
+                    } catch (e: Throwable) {
+                        e.printStackTrace()
+                    }
+                }
+            } else {
+                logger.debug("Scrum backlog doesn't contain sprint which is ready to be completed")
+            }
+        } else {
+            logger.error("scrumBoardsMemory must contain ScrumBoardPage implementations")

```

Comment:
This runtime error can be expressed as a compile error. We can change `BoardPagesMemory.scrum` to `Collection<ScrumBoardPage>`.